### PR TITLE
Add feature flag 'feature_new_room_decoration_ui' and segrate legacy UI component

### PR DESCRIPTION
**Pull request by germain-gg**
Fixes https://github.com/vector-im/element-web/issues/25887

Adds a new labs flag `feature_new_room_decoration_ui` that will be useful during the development of https://github.com/vector-im/element-web/issues/25883

**Current UI changes:**

Screenshot 2023-07-31 at 16 54 02

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/components/views/rooms/LegacyRoomHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,822 @@
+/*
```
Comment (t3chguy) [inline src/components/views/rooms/LegacyRoomHeader.tsx:1]: Many violations of https://github.com/vector-im/element-web/blob/develop/code_style.md#react 11

> A component should only use CSS class names in line with the component name.

> When knowingly using a class name from another component, document it with a [comment](https://github.com/vector-im/element-web/blob/develop/code_style.md#comments).

Comment (germain-gg) [reply src/components/views/rooms/LegacyRoomHeader.tsx:1]: Thank you for raising that. Fixed.

File: src/components/views/right_panel/RoomHeaderButtons.tsx [hunk 2]
```diff
@@ -64,14 +64,14 @@ const UnreadIndicator: React.FC<IUnreadIndicatorProps> = ({ color }) => {
 
     const classes = classNames({
         mx_Indicator: true,
-        mx_RoomHeader_button_unreadIndicator: true,
+        mx_LegacyRoomHeader_button_unreadIndicator: true,
```
Comment (t3chguy) [inline src/components/views/right_panel/RoomHeaderButtons.tsx:67]: This implies this should be LegacyRoomHeaderButtons

File: src/components/views/right_panel/HeaderButton.tsx [hunk 3]
```diff
@@ -45,9 +45,9 @@ export default class HeaderButton extends React.Component<IProps> {
         const { isHighlighted, isUnread = false, onClick, name, title, ...props } = this.props;
 
         const classes = classNames({
-            "mx_RoomHeader_button": true,
-            "mx_RoomHeader_button--highlight": isHighlighted,
-            "mx_RoomHeader_button--unread": isUnread,
+            "mx_LegacyRoomHeader_button": true,
```
Comment (t3chguy) [inline src/components/views/right_panel/HeaderButton.tsx:48]: This implies this should be LegacyHeaderButton

File: src/components/structures/RoomView.tsx [hunk 4]
```diff
@@ -345,22 +350,26 @@ function LocalRoomCreateLoader(props: ILocalRoomCreateLoaderProps): ReactElement
...
-                    e2eStatus={props.localRoom.encrypted ? E2EStatus.Normal : undefined}
-                    onAppsClick={null}
-                    appsShown={false}
-                    excludedRightPanelPhaseButtons={[]}
-                    showButtons={false}
-                    enableRoomOptionsMenu={false}
-                    viewingCall={false}
-                    activeCall={null}
-                />
+                {SettingsStore.getValue("feature_new_room_decoration_ui") ? (
```
Comment (t3chguy) [inline src/components/structures/RoomView.tsx:353]: Could we monitor/watch the setting instead of requiring a refresh?

**Review by t3chguy — COMMENTED** [review]
Github is making the RoomHeader/LegacyRoomHeader thing impossible to review. Any chance you could try to convince it that RoomHeader is a new file and that LegacyRoomHeader is predominantly a move? Maybe we could forgo most of this change by having `NewRoomHeader` until such time when we replace the "Legacy" one and thus everything is straight renames/moves. 

image

File: src/components/structures/RoomView.tsx (hunk 4 above)
Comment (germain-gg) [reply src/components/structures/RoomView.tsx:353]: I'd rather skip this change altogether, i do not want to add more logic that i need to write tests for.

File: src/components/views/right_panel/RoomHeaderButtons.tsx (hunk 2 above)
Comment (germain-gg) [reply src/components/views/right_panel/RoomHeaderButtons.tsx:67]: Sounds good, will update the files name. You're right.

File: src/components/views/right_panel/HeaderButton.tsx (hunk 3 above)
Comment (germain-gg) [reply src/components/views/right_panel/HeaderButton.tsx:48]: Sounds good, will update the files name. You're right.

**Comment by germain-gg** [comment]
Unfortunately no. I have not changed any of the logic.

File: src/components/views/right_panel/HeaderButton.tsx (hunk 3 above)
Comment (germain-gg) [reply src/components/views/right_panel/HeaderButton.tsx:48]: Actually, these classes should not be existing in this file in the first place. I will be keeping this file with the current name and will not spend time refactoring code we're looking to remove.

**Comment by t3chguy** [comment]
The diff disagrees. Could you rebase into sensible commits?

Maybe something like

1. Rename existing stuff
2. Create new RoomHeader

**Comment by t3chguy** [comment]
@germain-gg I don't think that quite worked - https://github.com/matrix-org/matrix-react-sdk/pull/11345/commits/b99467be705bf41ef71b38872fc32d711466a573 doesn't rename as it claims - it copies. Making the diff have an extra ~819 lines to review which is making it more difficult

https://github.com/matrix-org/matrix-react-sdk/pull/11345/commits/375904207884aca9c1ef526be43f07ee4ac99fd1 also doesn't create the new room header as it claims, instead it just modifies an existing file.

**Comment by germain-gg** [comment]
Well, i'm not sure how to make git change that. Further, but happy to hear suggestions

**Comment by t3chguy** [comment]
In https://github.com/matrix-org/matrix-react-sdk/commit/b99467be705bf41ef71b38872fc32d711466a573 you create a new file instead of renaming, not sure how else to say how to fix it. A rename != copy, its a move

**Comment by t3chguy** [comment]
@germain-gg like https://github.com/matrix-org/matrix-react-sdk/commit/cfef89227b7b61bc3a52f4c255995fe7870d0a26 + https://github.com/matrix-org/matrix-react-sdk/commit/4ff5ae96842a3861813be611c1b66f11a3786b27 but also for the pcss file

**Comment by germain-gg** [comment]
@t3chguy https://github.com/matrix-org/matrix-react-sdk/pull/11345/commits/f1aab65a1cfbed784f6f4fd8eb80d24144c0413e

**Comment by t3chguy** [comment]
@germain-gg thanks so much, those 2 commits look great. Just add one which fixes whatever is upsetting CI - don't worry about what it touches - and once that last commit looks good I'll approve

**Comment by germain-gg** [comment]
done. the commits contain the diff for the file rename.

**Comment by t3chguy** [comment]
@germain-gg while we wait for CI can you add it to labs.md?

**Pull request merged**